device/WakeOnLan/controller.js

```diff
@@ -0,0 +1,130 @@
+﻿'use strict';
+
+const BluePromise = require('bluebird');
+const fs = require('fs')
+const os = require('os');
+const net = require('net');
+const dgram = require('dgram');
+
+const PORT_NBR = 9;
+const DEFAULT_IPADDRESS = '255.255.255.255';
+
+let wolMap = "wol.map";
+
+/**
+ * Sets the path to the 'wol.map' file
+ */
+module.exports.setMapPath = function (mapPath) {
+    wolMap = mapPath;
+}
+
+/**
+ * Handles button presses for WOL
+ */
+module.exports.onButtonPressed = function (deviceid, name) {
+    console.log(`WOL button pressed for ${name}`);
+    var addr = parseAddr(name);
+
+    var macBuffer = getMacBuffer(addr.macAddress);
+
+    var magic = Buffer.alloc(6, 0xff);
+    for (var i = 0; i < 16; i++) {
+        magic = Buffer.concat([magic, macBuffer]);
+    }
+
+    var socket = dgram.createSocket(net.isIPv6(addr.ipAddress) ? 'udp6' : 'udp4')
+        .on('error', function (err) {
+            socket.close();
+            if (err !== null && err !== undefined) {
+                console.error("Exception creating socket", err);
+            }
+        })
+        .once('listening', function () {
+            socket.setBroadcast(true);
+        });
+
+    socket.send(magic, 0, magic.length, PORT_NBR, addr.ipAddress, function (err, res) {
+        socket.close();
+        if (err === null || err === undefined) {
+            console.log("Magic packet sent to device", name);
+        } else {
+            console.error("Exception writing the magic packet", err);
+        }
+    });
+}
+
+/**
+ * Discovers devices from the wolMap file.  Please note that validation of the
+ * address field is not 
+ */
+module.exports.discoverWolDevices = function () {
+    console.log('WOL discovery using %s', wolMap);
+    var contents = fs.readFileSync(wolMap, 'utf8');
+    var lines = contents.split('\n');
+    var devices = [];
+
+    var max = lines.length;
+    for (var i = 0; i < max; i++) {
+        var line = lines[i].endsWith('\r') ? lines[i].slice(0, -1) : lines[i];
```

Comment:
json would render this code obsolete, as you could access the attributes directly